test/erlfmt_SUITE.erl

```diff
@@ -1382,3 +1384,15 @@ overlong_warning(Config) when is_list(Config) ->
     ?assertEqual([], [Line || {8, _} = Line <- StringLongLines]),
     ?assert(lists:member({6, 121}, RangeLongLines)),
     ?assertEqual([], [Line || {8, _} = Line <- RangeLongLines]).
+
+do_not_crash_on_bad_record(Config) when is_list(Config) ->
+    %% bad record declaration
+    {ok, _, _} = erlfmt:read_nodes_string("nofile", "-record(r, [1])."),
+    %% bad record field
+    {ok, _, _} = erlfmt:read_nodes_string("nofile", "-record(r, {2})."),
+    %% bad record macro field
```

Comment:
this is actually valid code and the erlfmt_parse grammar allows macros in this position https://github.com/WhatsApp/erlfmt/blob/ec86fb280de822c3989bb14b28e7879f79532f7d/src/erlfmt_parse.yrl#L373

maybe something like this change would allow it to go through
```
- record_fields([{op,Am,'=',{atom,Aa,A},Expr}|Fields]) ->
-     [{record_field,Am,{atom,Aa,A},Expr}|record_fields(Fields)];
+ record_fields([{op,Am,'=',FieldName,Expr}|Fields]) ->
+    [{record_field,Am,FieldName,Expr}|record_fields(Fields)];
```

Reply:
I changed it back to `erl_anno:location` in `ret_err` and then tried this change, but it still broke.

The current pull request still returns the parse record as:
```
[{attribute,
      #{end_location => {1,21},location => {1,1}},
      {atom,
          #{end_location => {1,8},location => {1,2},text => "define"},
          define},
      [{var,#{end_location => {1,11},location => {1,9},text => "F1"},'F1'},
       {atom,
           #{end_location => {1,19},location => {1,13},text => "field1"},
           field1}]},
  {exprs,
      #{dot => true,end_location => {2,19},location => {2,1}},
      [{op,#{end_location => {2,18},location => {2,1}},
           '-',
           {call,
               #{end_location => {2,18},location => {2,2}},
               {atom,
                   #{end_location => {2,8},location => {2,2},text => "record"},
                   record},
               [{atom,
                    #{end_location => {2,10},location => {2,9},text => "r"},
                    r},
                {tuple,
                    #{end_location => {2,17},location => {2,12}},
                    [{macro_call,
                         #{end_location => {2,16},location => {2,13}},
                         {var,
                             #{end_location => {2,16},
                               location => {2,14},
                               text => "F1"},
                             'F1'},
                         none}]}]}}]}]
                         ```
                         
which I think is good enough
Parsing it as a `record_field` might presume too much, I mean technically it could be defining two record fields.
We do not necessarily want to format it was `record_field` in this case formatting it as a `macro_call` should be fine.

Reply:
I'm afraid that parsing as `exprs` will fail when using more specific record syntax. In particular something like `-record(foo, {?FOO, b :: any()})`.

I think parsing it as `record_field` with `?FOO` as the name makes sense. Yes, it's possible there's more expressions hidden there, but that shouldn't impact how we parse it, as long as it looks like a single expression.

Reply:
sorry, my diff suggestion was incorrect/incomplete, and I also didn't know that the parser would return an `exprs` element (I assumed it just gives up and returns an error)
it's up to you how you handle these scenarios

Reply:
I updated it with @gomoripeti 's suggestion

Reply:
Ah sorry, I see we want some of these bad records to actually parse as `{attribute ...` instead of `{exprs....`